AnimationSmoothness/explainer.md

```diff
@@ -0,0 +1,175 @@
+# A More Precise Way to Measure Animation Smoothness
+
+[comment]: < ** (*Same-Origin*) >
+
+Authors: [Jenna Sasson](https://github.com/jenna-sasson)
+
+## Status of this Document
+This document is a starting point for engaging the community and standards bodies in developing collaborative solutions fit for standardization. As the solutions to problems described in this document progress along the standards-track, we will retain this document as an archive and use this section to keep the community up-to-date with the most current standards venue and content location of future work and discussions.
+* This document status: **Active**
+* Expected venue: [W3C Web Performance Working Group](https://www.w3.org/groups/wg/webperf/)
+* **Current version: this document**
+
+## Introduction
+Smooth web animation is essential for a positive user experience. To understand the user’s experience with animation, quantifying their experience is an important initial step that allows web and browser developers to optimize their pages/engines and generate a more pleasing user experience. 
+
+Various metrics have been used in the past to try and understand the user’s experience in this space. Some of these were accessible to the webpage, while others were internal browser metrics. Examples of these include:
+
+* Framerate – the number of frames displayed to the user over time. 
```

Comment:
I think there may be security/privacy implications here, because the presence of a presented frame reveals new information to javascript. For example, imagine a root page (A) with a cross-origin login iframe (B): script on A can listen to the timing of the frames produced by B, and infer information about the password from the timing of the user's keystrokes. Another scenario is spellcheck: a page can toggle the spellcheck field of an input element and learn whether a word is misspelled based on the presence of a presented frame (for showing the spelling underline), and this contains information about the user's spellcheck dictionary. This information is not in requestAnimationFrame today because requesting a frame causes a new frame to be presented.

Reply:
Sure! https://github.com/MicrosoftEdge/MSEdgeExplainers/issues/1164